### ISSUE TITLE
File does not contain a valid CIL image.

### ISSUE THREAD

**Issue by JesperKuipers** [MCCTeam/Minecraft-Console-Client#2106]
Probably making a big beginners issue, but cant figure out how to get the binary file working.
Downloaded by putting  [this](https://github.com/MCCTeam/Minecraft-Console-Client/releases/download/20220816-28/MinecraftClient-linux.zip) link into the chromium browser. After that I unpacked it and used `sudo mono MinecraftClient`.
Instead of it working I get the following message:
![afbeelding](https://user-images.githubusercontent.com/44586411/184984902-8ad57972-8794-471d-9022-aa0547e94438.png)

And yes I used the things below:
`On Linux: sudo apt-get install mono-runtime libmono-reflection-cil Then, open a terminal in this folder and run "mono MinecraftClient.exe". If you cannot authenticate on Mono because you have TLS/HTTPS/Certificate errors, you'll need to run mozroots --import --ask-remove once or install ca-certificates-mono (See https://github.com/MCCTeam/Minecraft-Console-Client/issues/1708#issuecomment-893768862). If Mono crashes, retry with mono-complete instead of mono-runtime. Use at least Mono v4.0.`

**Comment by milutinke** [comment]
Could be that it was not built correctly.
Try building it yourself.
Follow the steps from: https://mccteam.github.io/docs/guide/installation.html

**Comment by JesperKuipers** [comment]
Can't get .net to work so idk :P

**Comment by JesperKuipers** [comment]
@milutinke tried to build it on windows, got this 
![afbeelding](https://user-images.githubusercontent.com/44586411/184988384-48a061ff-a8bb-4f1a-be8d-dd4724792a63.png)

**Comment by ORelio** [comment]
We're in the process of moving away from .NET 4.0, so the process for .NET 6 (newer development builds) is the following:
1) Install .NET, here are 2 Microsoft documentation pages for Linux and Raspberry Pi:
    - https://docs.microsoft.com/en-us/dotnet/core/install/linux
    - https://docs.microsoft.com/en-us/dotnet/iot/deployment
2) Launch the executable directly: `./MinecraftClient`

You no longer need to use Mono to launch the program on Mac and Linux, but instead install .NET like when using Windows.

**Comment by milutinke** [comment]
> We're in the process of moving away from .NET 4.0, so the process for .NET 6 (newer development builds) is the following:
> 
>     1. Install .NET, here are 2 Microsoft documentation pages for Linux and Raspberry Pi:
>        
>        * https://docs.microsoft.com/en-us/dotnet/core/install/linux
>        * https://docs.microsoft.com/en-us/dotnet/iot/deployment
> 
>     2. Launch the executable directly: `./MinecraftClient`
> 
> 
> You no longer need to use Mono to launch the program on Mac and Linux, but instead install .NET like when using Windows.

We should update the documentation.

**Comment by milutinke** [comment]
> @milutinke tried to build it on windows, got this ![afbeelding](https://user-images.githubusercontent.com/44586411/184988384-48a061ff-a8bb-4f1a-be8d-dd4724792a63.png)

Updated the documentation for .NET 6.0, try it out now.

**Comment by JesperKuipers** [comment]
@milutinke it isnt updated in the [readme](https://github.com/MCCTeam/Minecraft-Console-Client/tree/master/MinecraftClient/config#minecraft-console-client-user-manual). I would suggets either putting it in both or replace all the text in the readme for a link to the website

**Comment by milutinke** [comment]
> @milutinke it isnt updated in the [readme](https://github.com/MCCTeam/Minecraft-Console-Client/tree/master/MinecraftClient/config#minecraft-console-client-user-manual). I would suggets either putting it in both or replace all the text in the readme for a link to the website

Can't update the repository until #1943 is merged.

**Comment by JesperKuipers** [comment]
> We're in the process of moving away from .NET 4.0, so the process for .NET 6 (newer development builds) is the following:
> 
>     1. Install .NET, here are 2 Microsoft documentation pages for Linux and Raspberry Pi:
>        
>        * https://docs.microsoft.com/en-us/dotnet/core/install/linux
>        * https://docs.microsoft.com/en-us/dotnet/iot/deployment
> 
>     2. Launch the executable directly: `./MinecraftClient`
> 
> 
> You no longer need to use Mono to launch the program on Mac and Linux, but instead install .NET like when using Windows.

I can build my own version with ur method, but cant seem to run the file. *cannot execute binary file: Exec format error*
![afbeelding](https://user-images.githubusercontent.com/44586411/185255165-3b1b7f02-6edb-4f54-a336-22fa366a51b6.png)

**Comment by JesperKuipers** [comment]
Btw im running raspian 64 bit on a ARM cpu

**Comment by milutinke** [comment]
> Btw im running raspian 64 bit on a ARM cpu ![afbeelding](https://user-images.githubusercontent.com/44586411/185257570-35468b15-bd9a-4bb3-9d32-6a0c6149f04e.png)

Oh, that is why it does not work xD
The build process is written for i386 64 bit.
Use the following build command for Linux on Arm 64 bit:
```bash
dotnet publish MinecraftClient -f net6.0 -r linux-arm64 --no-self-contained -c Release -p:UseAppHost=true -p:IncludeNativeLibrariesForSelfExtract=true -p:DebugType=None
```

**Comment by milutinke** [comment]
I've updated the Documentation to include other platforms than i386 64 bit.

**Comment by JesperKuipers** [comment]
@milutinke thanks, im 1 step closer now :) 
![afbeelding](https://user-images.githubusercontent.com/44586411/185359997-34a629fc-919b-47b4-a91b-633b485f6fc9.png)

**Comment by JesperKuipers** [comment]
browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"

**Comment by milutinke** [comment]
> browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"

Can you test this for me if it is not a problem?

In https://github.com/MCCTeam/Minecraft-Console-Client/blob/a8bbb1ac766613952e5f9bff1b3eab9e709b182c/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L90
Replace the method `OpenBrowser` with the new one bellow:
```csharp
public static void OpenBrowser(string link)
{
	try
	{
		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
		{
			link = link.Replace("&", "^&");
			Process.Start(new ProcessStartInfo(link) { UseShellExecute = true });
		}
		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
		{
			Process.Start("xdg-open", link);
		}
		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
		{
			Process.Start("open", link);
		}
		else
		{
			throw new Exception("Platform not supported, open up the link manually: " + link);
		}
	}
	catch (Exception e)
	{
		ConsoleIO.WriteLine("Cannot open browser\n" + e.Message + "\n" + e.StackTrace);
	}
}
```

And build the project and test if it opens the web browser now.

**Comment by milutinke** [comment]
> browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"

Can you enable debug messages and provide the log (Make sure to redact your auth token)

**Comment by JesperKuipers** [comment]
> > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> 
> Can you enable debug messages and provide the full log (Make sure to redact your auth token)

Sorry how do enable debug again?

**Comment by milutinke** [comment]
> > > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> > 
> > Can you enable debug messages and provide the full log (Make sure to redact your auth token)
> 
> Sorry how do I enable debug again?

https://mccteam.github.io/docs/guide/configuration.html#debugmessages

Set it to `true`

**Comment by JesperKuipers** [comment]
> > > > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> > > 
> > > 
> > > Can you enable debug messages and provide the full log (Make sure to redact your auth token)
> > 
> > 
> > Sorry how do I enable debug again?
> 
> https://mccteam.github.io/docs/guide/configuration.html#debugmessages
> 
> Set it to `true`

Hope this helps

![afbeelding](https://user-images.githubusercontent.com/44586411/185389766-e77e7728-a971-4c2b-a13d-cb2e175dcd01.png)

**Comment by milutinke** [comment]
Something is wrong on the Microsoft side, I don't have an XBox Live account so I can't test, so you will have to add a debug line and screenshot the response from the XBox live service.

Bellow this line https://github.com/MCCTeam/Minecraft-Console-Client/blob/73195c55d4f0f041e744e5d923092655c2667185/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L138

Add: `Console.WriteLine(html);`
Build the project and run it, it should output html.

**Comment by ReinforceZwei** [comment]
`PPFT` and `urlPost` are values used for later sign-in. `Fail to extract PPFT or urlPost` mean MS probably updated their sign-in page so that the hardcoded regex cannot extract those value from HTML body anymore. 
https://github.com/MCCTeam/Minecraft-Console-Client/blob/73195c55d4f0f041e744e5d923092655c2667185/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L119-L120
The `mcc` sign-in were expected to be broken at some point. Maybe it is the time to deprecate the `mcc` sign-in method.

**Comment by JesperKuipers** [comment]
> `PPFT` and `urlPost` are values used for later sign-in. `Fail to extract PPFT or urlPost` mean MS probably updated their sign-in page so that the hardcoded regex cannot extract those value from HTML body anymore.
> 
> https://github.com/MCCTeam/Minecraft-Console-Client/blob/73195c55d4f0f041e744e5d923092655c2667185/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L119-L120
> 
> See my comment about `mcc` sign-in method here [#1379 (comment)](https://github.com/MCCTeam/Minecraft-Console-Client/issues/1379#issuecomment-749559312)
> 
> The `mcc` sign-in were expected to be broken at some point. Maybe it is the time to deprecate the `mcc` sign-in method.

so what do you recommend?

**Comment by JesperKuipers** [comment]
> Something is wrong on the Microsoft side, I don't have an XBox Live account so I can't test, so you will have to add a debug line and screenshot the response from the XBox live service. It could be that they have changed something, like a format of the returned value.
> 
> Bellow this line
> 
> https://github.com/MCCTeam/Minecraft-Console-Client/blob/73195c55d4f0f041e744e5d923092655c2667185/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L138
> 
> Add: `Console.WriteLine(html);` Build the project and run it, it should output html, paste it here. I do not know if the PPFT is important, but redact it if it is given

![afbeelding](https://user-images.githubusercontent.com/44586411/185408874-6a7d4f5f-efb3-4b2c-8765-3d172316e618.png)
gives this as output, consolwrite doesnt seem to be fired
![afbeelding](https://user-images.githubusercontent.com/44586411/185412769-43e926d8-4050-4ae7-a67a-9aa2b1175483.png)

**Comment by JesperKuipers** [comment]
> > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> 
> Can you test this for me if it is not a problem?
> 
> In
> 
> https://github.com/MCCTeam/Minecraft-Console-Client/blob/a8bbb1ac766613952e5f9bff1b3eab9e709b182c/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L90
> 
> Replace the method `OpenBrowser` with the new one bellow:
> 
> ```cs
> public static void OpenBrowser(string link)
> {
> 	try
> 	{
> 		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
> 		{
> 			link = link.Replace("&", "^&");
> 			Process.Start(new ProcessStartInfo(link) { UseShellExecute = true });
> 		}
> 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
> 		{
> 			Process.Start("xdg-open", link);
> 		}
> 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
> 		{
> 			Process.Start("open", link);
> 		}
> 		else
> 		{
> 			throw new Exception("Platform not supported, open up the link manually: " + link);
> 		}
> 	}
> 	catch (Exception e)
> 	{
> 		ConsoleIO.WriteLine("Cannot open browser\n" + e.Message + "\n" + e.StackTrace);
> 	}
> }
> ```
> 
> And build the project and test if it opens the web browser now.

Gives a shit ton of errors 
![afbeelding](https://user-images.githubusercontent.com/44586411/185415757-9580a960-77a2-495c-b828-7a3441cda631.png)

**Comment by milutinke** [comment]
> > > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> > 
> > 
> > Can you test this for me if it is not a problem?
> > In
> > https://github.com/MCCTeam/Minecraft-Console-Client/blob/a8bbb1ac766613952e5f9bff1b3eab9e709b182c/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L90
> > 
> > Replace the method `OpenBrowser` with the new one bellow:
> > ```cs
> > public static void OpenBrowser(string link)
> > {
> > 	try
> > 	{
> > 		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
> > 		{
> > 			link = link.Replace("&", "^&");
> > 			Process.Start(new ProcessStartInfo(link) { UseShellExecute = true });
> > 		}
> > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
> > 		{
> > 			Process.Start("xdg-open", link);
> > 		}
> > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
> > 		{
> > 			Process.Start("open", link);
> > 		}
> > 		else
> > 		{
> > 			throw new Exception("Platform not supported, open up the link manually: " + link);
> > 		}
> > 	}
> > 	catch (Exception e)
> > 	{
> > 		ConsoleIO.WriteLine("Cannot open browser\n" + e.Message + "\n" + e.StackTrace);
> > 	}
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And build the project and test if it opens the web browser now.
> 
> Gives a shit ton of errors ![afbeelding](https://user-images.githubusercontent.com/44586411/185415757-9580a960-77a2-495c-b828-7a3441cda631.png)

Those are warnings, they're there since the port to .NET 5.0, we're still to deal with them.

**Comment by JesperKuipers** [comment]
> > > > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> > > 
> > > 
> > > Can you test this for me if it is not a problem?
> > > In
> > > https://github.com/MCCTeam/Minecraft-Console-Client/blob/a8bbb1ac766613952e5f9bff1b3eab9e709b182c/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L90
> > > 
> > > Replace the method `OpenBrowser` with the new one bellow:
> > > ```cs
> > > public static void OpenBrowser(string link)
> > > {
> > > 	try
> > > 	{
> > > 		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
> > > 		{
> > > 			link = link.Replace("&", "^&");
> > > 			Process.Start(new ProcessStartInfo(link) { UseShellExecute = true });
> > > 		}
> > > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
> > > 		{
> > > 			Process.Start("xdg-open", link);
> > > 		}
> > > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
> > > 		{
> > > 			Process.Start("open", link);
> > > 		}
> > > 		else
> > > 		{
> > > 			throw new Exception("Platform not supported, open up the link manually: " + link);
> > > 		}
> > > 	}
> > > 	catch (Exception e)
> > > 	{
> > > 		ConsoleIO.WriteLine("Cannot open browser\n" + e.Message + "\n" + e.StackTrace);
> > > 	}
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And build the project and test if it opens the web browser now.
> > 
> > 
> > Gives a shit ton of errors ![afbeelding](https://user-images.githubusercontent.com/44586411/185415757-9580a960-77a2-495c-b828-7a3441cda631.png)
> 
> Those are warnings, they're there since the port to .NET 5.0, we're still to deal with them. It should build, you just need to include: `using System.Runtime.InteropServices;` at the top of the file, I forgot to mention that.

Thanks, the browser opens automatic now.
This happens when i copy paste the key from the site 
![afbeelding](https://user-images.githubusercontent.com/44586411/185439802-e8236ced-8ea9-4bc1-82ba-16933f7a9f9e.png)

**Comment by milutinke** [comment]
Check to see if you're not rate limited by Microsoft by any chance, I know that i has happened to some people.
The weird thing is that I do not see the Response information.

Check out the discussion
https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756

I think it's a Microsoft related problem, maybe they're having issues with their services, or your packets are getting dropped/blocked on the way.

**Comment by JesperKuipers** [comment]
> Check to see if you're not rate limited by Microsoft by any chance, I know that i has happened to some people. The weird thing is that I do not see the Response information.
> 
> Check out the discussion [#2020 (comment)](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756)
> 
> I think it's a Microsoft related problem, maybe they're having issues with their services, or your packets are getting dropped/blocked on the way.
> 
> Also, maybe try his [code](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020/files) from the commit, it might work.

prob doing something wrong, messed with it for some time.
![afbeelding](https://user-images.githubusercontent.com/44586411/185487845-2bce6f9e-30fa-4421-848b-4f5c679a12d5.png)

**Comment by JesperKuipers** [comment]
> > Check to see if you're not rate limited by Microsoft by any chance, I know that i has happened to some people. The weird thing is that I do not see the Response information.
> > Check out the discussion [#2020 (comment)](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756)
> > I think it's a Microsoft related problem, maybe they're having issues with their services, or your packets are getting dropped/blocked on the way.
> > Also, maybe try his [code](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020/files) from the commit, it might work.
> 
> prob doing something wrong, messed with it for some time. ![afbeelding](https://user-images.githubusercontent.com/44586411/185487845-2bce6f9e-30fa-4421-848b-4f5c679a12d5.png) ![afbeelding](https://user-images.githubusercontent.com/44586411/185487904-6d584889-43e0-4def-860b-bd140a79e3cb.png)

fixed it https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020/files#r949591756

**Comment by milutinke** [comment]
Replace `{Exception e}` with `(Exception e)`

**Comment by JesperKuipers** [comment]
> Replace `{Exception e}` with `(Exception e)` Edit; Oh, you've fixed it.

got new issue :P https://github.com/MCCTeam/Minecraft-Console-Client/issues/2106#issuecomment-1219958561

**Comment by milutinke** [comment]
> > Replace `{Exception e}` with `(Exception e)` Edit; Oh, you've fixed it.
> 
> got new issue :P ![afbeelding](https://user-images.githubusercontent.com/44586411/185494125-675a70c3-1b2c-47ce-8994-22c204f89220.png)

I've actually read through the code, it just catches an exception, doesn't do anything to fix the problem.
The main root issue is with the Microsoft.
Try using a VPN or proxy to login, it may help.

**Comment by JesperKuipers** [comment]
> > > Replace `{Exception e}` with `(Exception e)` Edit; Oh, you've fixed it.
> > 
> > 
> > got new issue :P ![afbeelding](https://user-images.githubusercontent.com/44586411/185494125-675a70c3-1b2c-47ce-8994-22c204f89220.png)
> 
> I've actually read through the code, it just catches an exception, doesn't do anything to fix the problem. The main root issue is with the Microsoft. Try using a VPN or proxy to login, it may help.
> 
> Or try dumping what is in the response.

cant build now tho, so revered https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756 ?

**Comment by milutinke** [comment]
> > > > Replace `{Exception e}` with `(Exception e)` Edit; Oh, you've fixed it.
> > > 
> > > 
> > > got new issue :P ![afbeelding](https://user-images.githubusercontent.com/44586411/185494125-675a70c3-1b2c-47ce-8994-22c204f89220.png)
> > 
> > 
> > I've actually read through the code, it just catches an exception, doesn't do anything to fix the problem. The main root issue is with the Microsoft. Try using a VPN or proxy to login, it may help.
> > Or try dumping what is in the response.
> 
> cant build now tho, so revered [#2020 (comment)](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756) ?

I haven't seen that it was closed.
Do a dump like this, and sceenshot, redact `auth_token` if it there.
![image](https://user-images.githubusercontent.com/441903/185495533-030b0f77-20ec-4b37-b6eb-4a90f3d5089c.png)

**Comment by JesperKuipers** [comment]
@milutinke i can screen share on discord if u want

**Comment by milutinke** [comment]
> @milutinke i can screen share on discord if u want

Can't Discord right now, it's late, tomorrow I'll most likely not be on the pc.
Try the code above when you can and slap the results here.
If it doesn't work, we're gonna literally add debug lines everywhere and output everything into a file until we figure out what is actually happening.

**Comment by JesperKuipers** [comment]
> > @milutinke i can screen share on discord if u want
> 
> Can't Discord right now, it's late, tomorrow I'll most likely not be on the pc. Try the code above when you can and slap the results here. If it doesn't work, we're gonna literally add debug lines everywhere and output everything into a file until we figure out what is actually happening.

![afbeelding](https://user-images.githubusercontent.com/44586411/185497173-3dc57a78-921c-4e36-bc68-8ba66aaa23a3.png)
:'(

**Comment by JesperKuipers** [comment]
would prefer mcc to work instead of browser tho

**Comment by JesperKuipers** [comment]
it worked btw on the account, but then found out could not connect to any server without migrating to MS account, since then the issue started

**Comment by milutinke** [comment]
> it worked btw on the account, but then found out could not connect to any server without migrating to MS account, since then the issue started. But my main account is already migrated and had/has the same issues. So I suspect its something on MS their side which fucks it up with MS accounts but not with Mojang accounts

Is it working on Windows by any chance?
I haven't booted my Arch for a couple of days because I do not have dotnet env there, I'll try it out tomorrow on Arch.

**Comment by JesperKuipers** [comment]
> > it worked btw on the account, but then found out could not connect to any server without migrating to MS account, since then the issue started. But my main account is already migrated and had/has the same issues. So I suspect its something on MS their side which fucks it up with MS accounts but not with Mojang accounts
> 
> Is it working on Windows by any chance? I haven't booted my Arch for a couple of days because I do not have dotnet env there, I'll try it out tomorrow on Arch.

latest release doesnt work on windows for me 
![afbeelding](https://user-images.githubusercontent.com/44586411/185499770-a147bfad-0f14-4e6a-834e-64760c2ba3ee.png)

**Comment by milutinke** [comment]
> > > it worked btw on the account, but then found out could not connect to any server without migrating to MS account, since then the issue started. But my main account is already migrated and had/has the same issues. So I suspect its something on MS their side which fucks it up with MS accounts but not with Mojang accounts
> > 
> > 
> > Is it working on Windows by any chance? I haven't booted my Arch for a couple of days because I do not have dotnet env there, I'll try it out tomorrow on Arch.
> 
> latest release doesnt work on windows for me ![afbeelding](https://user-images.githubusercontent.com/44586411/185499770-a147bfad-0f14-4e6a-834e-64760c2ba3ee.png)

Have you tried building from the source?
Also, which version of .NET are you running?

**Comment by kvr25565** [comment]
> I've updated the Documentation to include other platforms than i386 64 bit.

can you give me a link?

**Comment by milutinke** [comment]
> > I've updated the Documentation to include other platforms than i386 64 bit.
> 
> can you give me a link?

https://mccteam.github.io/docs/guide/installation.html

**Comment by milutinke** [comment]
Tested on Arch Intel x64, works normally.
I think it's a Microsoft problem.
So, try using a proxy just in case and try it on windows too.

**Comment by milutinke** [comment]
Submitted a fix for the browser not opening,
If everyone agrees, I'll close the issue since it's out of our control.

**Comment by JesperKuipers** [comment]
> Tested on Arch Intel x64, works normally. I think it's a Microsoft problem. So, try using a proxy just in case and try it on windows too. I'll do a commit to fix the Browser not opening when I have some time.

how would this be a MS issue? Doesn't work on 2 accounts. Can login with both accounts on the minecraft website, the minecraft launcher and the browser in mcc.

**Comment by milutinke** [comment]
> > Tested on Arch Intel x64, works normally. I think it's a Microsoft problem. So, try using a proxy just in case and try it on windows too. I'll do a commit to fix the Browser not opening when I have some time.
> 
> how would this be a MS issue? Doesn't work on 2 accounts. Can login with both accounts on the minecraft website, the minecraft launcher and the browser in mcc.

I've only been able to test the browser login, can't test the mcc login since I have 2FA. I've had 5 of my friends test the browser login on various platforms and it works.
As for the MCC login, I do not have a way to test it.
So the browser login works for you, but the MCC login doesn't?

**Comment by JesperKuipers** [comment]
> > > > > browser doesnt open, but the link works and I can copy the code, but if i paste it in mcc it will take a while and then give the message "killed"
> > > > 
> > > > 
> > > > Can you test this for me if it is not a problem?
> > > > In
> > > > https://github.com/MCCTeam/Minecraft-Console-Client/blob/a8bbb1ac766613952e5f9bff1b3eab9e709b182c/MinecraftClient/Protocol/MicrosoftAuthentication.cs#L90
> > > > 
> > > > Replace the method `OpenBrowser` with the new one bellow:
> > > > ```cs
> > > > public static void OpenBrowser(string link)
> > > > {
> > > > 	try
> > > > 	{
> > > > 		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
> > > > 		{
> > > > 			link = link.Replace("&", "^&");
> > > > 			Process.Start(new ProcessStartInfo(link) { UseShellExecute = true });
> > > > 		}
> > > > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
> > > > 		{
> > > > 			Process.Start("xdg-open", link);
> > > > 		}
> > > > 		else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
> > > > 		{
> > > > 			Process.Start("open", link);
> > > > 		}
> > > > 		else
> > > > 		{
> > > > 			throw new Exception("Platform not supported, open up the link manually: " + link);
> > > > 		}
> > > > 	}
> > > > 	catch (Exception e)
> > > > 	{
> > > > 		ConsoleIO.WriteLine("Cannot open browser\n" + e.Message + "\n" + e.StackTrace);
> > > > 	}
> > > > }
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > And build the project and test if it opens the web browser now.
> > > 
> > > 
> > > Gives a shit ton of errors ![afbeelding](https://user-images.githubusercontent.com/44586411/185415757-9580a960-77a2-495c-b828-7a3441cda631.png)
> > 
> > 
> > Those are warnings, they're there since the port to .NET 5.0, we're still to deal with them. It should build, you just need to include: `using System.Runtime.InteropServices;` at the top of the file, I forgot to mention that.
> 
> Thanks, the browser opens automatic now. This happens when i copy paste the key from the site ![afbeelding](https://user-images.githubusercontent.com/44586411/185439992-f84ae716-75b0-4cfb-8c49-76e627163cef.png)

this is with browser, after copy pasting the code from the browser

**Comment by milutinke** [comment]
> > > > > Replace `{Exception e}` with `(Exception e)` Edit; Oh, you've fixed it.
> > > > 
> > > > 
> > > > got new issue :P ![afbeelding](https://user-images.githubusercontent.com/44586411/185494125-675a70c3-1b2c-47ce-8994-22c204f89220.png)
> > > 
> > > 
> > > I've actually read through the code, it just catches an exception, doesn't do anything to fix the problem. The main root issue is with the Microsoft. Try using a VPN or proxy to login, it may help.
> > > Or try dumping what is in the response.
> > 
> > 
> > cant build now tho, so revered [#2020 (comment)](https://github.com/MCCTeam/Minecraft-Console-Client/pull/2020#issuecomment-1219410756) ?
> 
> I haven't seen that it was closed. Do a dump like this, and sceenshot, redact `auth_token` if it there. ![image](https://user-images.githubusercontent.com/441903/185495533-030b0f77-20ec-4b37-b6eb-4a90f3d5089c.png)
> 
> It's painful when I can't re-create the problem myself, for me it the browser method works like a charm.

@JesperKuipers Have you done this?

**Comment by JesperKuipers** [comment]
> > > > it worked btw on the account, but then found out could not connect to any server without migrating to MS account, since then the issue started. But my main account is already migrated and had/has the same issues. So I suspect its something on MS their side which fucks it up with MS accounts but not with Mojang accounts
> > > 
> > > 
> > > Is it working on Windows by any chance? I haven't booted my Arch for a couple of days because I do not have dotnet env there, I'll try it out tomorrow on Arch.
> > 
> > 
> > latest release doesnt work on windows for me ![afbeelding](https://user-images.githubusercontent.com/44586411/185499770-a147bfad-0f14-4e6a-834e-64760c2ba3ee.png)
> 
> Have you tried building from the source? Also, which version of .NET are you running?

was on net5, was using visual studio 2019. upgraded to VS 2022 (which has net6) and got it working.
![afbeelding](https://user-images.githubusercontent.com/44586411/185911250-5ee7b668-41bf-431b-9fff-668474ffd232.png)

So yes it is something in the code. Dm me on discord (itz_game_pro#9999) if you need anything from me. Could port forward my vnc server on the RaspberryPi so you can check it for yourself if u want.

**Comment by milutinke** [comment]
The issue was on the local network.
Closing down the issue.
